src/main/scala/edu/ie3/powerFactory2psdm/converter/NodeConverter.scala

```diff
@@ -83,6 +97,70 @@ object NodeConverter {
     )
   }
 
+  /** Creates a Map that maps from unsafe node ids to uuids from a csv file that
+    * is used for node conversion. This can be used to keep uuids of the nodes
+    * consistent between original data and the converted grid.
+    *
+    * @param nodeUuidMappingInformation
+    *   necessary mapping information
+    * @return
+    *   Map from unsafe node id to uuid
+    */
+  def getNodeNameMapping(
+      nodeUuidMappingInformation: NodeUuidMappingInformation
+  ): Map[String, UUID] = {
+    // parse csv file
+    val bufferedSource = Source.fromFile(nodeUuidMappingInformation.filePath)
+    val lines = bufferedSource.getLines()
+
+    lines.take(1).next.split(nodeUuidMappingInformation.csvSeparator) match {
+      case Array("uuid", "id") =>
+      case _ =>
+        throw new IOException(
+          "Invalid CSV header. We expect the header of the node name mapping file to be \"uuid\" [CSV Sep.] \"id\"."
+        )
+    }
+
+    val idsAndUuids = lines.map { line =>
+      line
+        .split(nodeUuidMappingInformation.csvSeparator)
+        .map(_.trim) match {
+        case Array(uuidString, id) =>
+          val uuid = Try(UUID.fromString(uuidString)) match {
+            case Failure(exception) =>
+              throw new IllegalArgumentException(
+                s"UUID: $uuidString on line: $line is not a valid UUID. Reason: ${exception.getMessage}"
+              )
```

Comment:
Include caught exception when throwing new exception 

```suggestion
              throw new IllegalArgumentException(
                s"UUID: $uuidString on line: $line is not a valid UUID.",
                exception
              )
```